app/uk/gov/hmrc/agentsubscriptionfrontend/controllers/CheckAgencyController.scala

```diff
@@ -184,18 +186,35 @@ class CheckAgencyController @Inject()(
         }
       }
 
-    agentSubscriptionConnector.getRegistration(knownFacts.utr, knownFacts.postcode) flatMap {
-      case Some(Registration(Some(taxpayerName), isSubscribedToAgentServices)) if !isSubscribedToAgentServices =>
-        processCheckAgencyStatus(knownFacts.utr, taxpayerName, isSubscribedToAgentServices)
-      case Some(Registration(_, isSubscribedToAgentServices)) if isSubscribedToAgentServices =>
+    subscriptionService.getSubscriptionStatus(knownFacts.utr, knownFacts.postcode).flatMap {
+      case SubscriptionProcess(SubscriptionState.BrandNewSubscription, Some(registrationDetails)) =>
+        processCheckAgencyStatus(
+          knownFacts.utr,
+          registrationDetails.taxpayerName.get,
+          registrationDetails.isSubscribedToAgentServices)
+      case SubscriptionProcess(SubscriptionState.IsOnlySubscribedInETMP, Some(registrationDetails)) =>
+        agent match {
+          case hasNonEmptyEnrolments(_) =>
+            Future successful Redirect(routes.CheckAgencyController.showHasOtherEnrolments())
+          case _ => {
+            subscriptionService
+              .completePartialSubscription(
+                CompletePartialSubscriptionBody(knownFacts.utr, SubscriptionRequestKnownFacts(knownFacts.postcode)))
+              .map {
+                case (_, Some(arn)) =>
+                  Redirect(routes.SubscriptionController.showSubscriptionComplete()).flashing("arn" -> arn.value)
```

Comment:
I noticed there's a few metrics on the other outcomes in this endpoint - maybe we could add something like `mark("Count-Subscription-PartialSubscriptionCompleted")` to keep it consistent and complete?

Reply:
yup, done